### PR TITLE
[Experimental] Product Filters: New and improved blocks structure

### LINKED ISSUE #50867: Product Filters Redesign: Convert List style into an inner block for sharing between filter blocks
## Why do we need this?

The new design introduces the settings for list and chip styles. Which raises a need for better component sharing.

image

We created a new concept called the Interactivity component to share a List on the front end. But we don't have the same thing for the editor. I [tried](https://github.com/woocommerce/woocommerce/pull/50362) to add sharable block settings for the Interactivity editor component but it feels quirky and hassle to use.

Block settings should belong to blocks, so I think we should at least try to turn the list into an inner block for any filter blocks using the List style. Similar to https://github.com/woocommerce/woocommerce/issues/50871, we need to display the List settings in parent blocks like Filter Options or Product Filter.

The work on this will influence #49975.

### LINKED ISSUE #50871: Product Filters Redesign: Display Filter Options setting in Product Filter wrapper block
When users select a wrapper block like the Color block, they don't see any settings because the block is just a wrapper. They need to select the filter options block to see the settings, which is confusing.

In the design, the structure of a Filter block, take 3td38XZM4P9WhWg618nXjt-fi-334_124862 as the example, is:
- Price: The wrapper block and filter settings live here.
    - Header
        - Heading
        - Clear button
    - Filter Options: the block that renders filter options.
In reality, we ended up with a different structure, which is inherited from the previous iteration of this project.
- Price: The variation of the wrapper block (woocommerce/product-filter). We have other variations for other filters like Stock, Rating, and each product attribute (Color, Size, ...).
    - Header
        - Heading
        - Clear button
    - Filter Options: the actual Price Filter block that holds all block attributes, block settings, and render filter options.
There are some reasons we did it that way:
- We inherit the structure from legacy filter blocks.
- We use variation for wrapper because it's a good usage of variation, we don't have to maintain multiple identical blocks.
But this approach also creates an issue we're talking about in this thread. The filter block settings live in the Filter Options block, while the wrapper block doesn't have any settings, which raises the need for sharing block settings from the inner block to the wrapper block.

If we go with creating blocks for each display style (#50867), those blocks have settings too. Users shouldn't have to select all the way down to access block settings, instead, we should make those settings available in the wrapper block as well as the Filter Options block.

### PR DESCRIPTION
**Pull request by dinhtungdu**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #50871,  closes #50867.

This PR introduces a new block structure as proposed in [https://github.com/woocommerce/woocommerce/issues/50871#issuecomment-2320933584](https://github.com/woocommerce/woocommerce/issues/50871#issuecomment-2320933584):

- Removes the Product Filter block (we keep the code for now, but it will be eventually removed from the source code).
- Updates the Attribute Filter block:
    - Makes it a parent block like the `core/group` block, which contains the Header and the new List block.
    - Moves the rendering logic to the List block. 
    - Passes the filter options down to the List block through context.
- Creates a new List block.
    - Renders the filter options as a list from the data passed through block context.
    - Have advanced settings to customize the list further ~(WIP)~
    - Have improved block select behavior:
        - When users select the block from the canvas by clicking on the filter options preview, the Attribute Filter block is selected.
        - When the Attribute Filter block is selected, users can click on the filter options preview on more time to select the List block.
        - The outline will inform users about the currently selected block.
        - The block select behavior inside the List view are same as other blocks.
- Adds Interactivity API directives to the `Product Filters` block.
- Updates the Product Filters template part, the template now contains only Active and Attribute blocks.
- Note that the Active block hasn't been updated to the new structure yet. 

#### Why should we use the new structure?

For more details, please check #50871. Here is a TL;DR:

- A simple structure that follows the WP Core, where the filter block acts as the `core/query` block, but with some filter settings like attribute selector.
- Avoid hacking the editor by sharing the block settings from the inner block to the parent block.
- Split the responsibility to increase the reusability:
    - The group block (filter blocks) holds the header and filter options. Compared to the `core/group` block, the filter block has some additional responsibilities:
        - Processing the filter data and passing it down through the context for the inner block to consume. 
        - Rendering and updating inner blocks based on the display style setting.
    - The List/Chips/Slider block renders the filter options:
        - Each display style has a corresponding block.
        - The block can be used among multiple filter blocks:
            - List: Attribute, Stock, Rating
            - Chips: Attribute, Active.
        - This paves the way for us and third parties to create more display styles like Color Swatches, Image Swatches, Size Grid, Price List, etc.

image

#### Notes:
- The price, stock, and rating filter blocks are removed from the template part as they need updating to remove the wrapper. We will add them back when working on the new design of each block later.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure the experimental blocks are enabled:
    - If you use the WooCommerce Beta Tester plugin, you can enable the experimental blocks in Tools > WCA Test Helper > Features. Toggle the `experimental-blocks` option.
    - If you build this PR locally, build the branch with `pnpm --filter='@woocommerce/plugin-woocommerce' watch:build` command to have experimental blocks available.
2. Reset any customization made to the Product Filters template part.
3. Add the template part to the Product Catalog template.
4. Save and check the Shop page on the front end. See:
    1. The Attribute Filter block picks up the correct block to render, with the correct title set to the product attribute label.
    2. Selecting one attribute, see the selected attribute appears in the Active block.
    3. See the Product Collection block updates.
    4. See the URL updates.
5. Edit the Product Catalog template or Product Filters template part:
    1. See the preview renders Active and Attribute filter block.
    2. Open the List view and Sidebar.
    3. From the canvas, click on the filter options. See the Attribute Filter block is selected, not the List block.
    4. Try changing the block setting, see the preview updates.
    6. Click on the filter options again, and see the List block is selected.
    7. Change the color settings of the List block, see the editor preview update as you change.
    8. Chang the display style from List to Chip. See the editor preview change. The Chips block hasn't been worked on, so you will see the placeholder.
    9. Change back to the List block, see the customization you did previously is preserved.
    10. Save the template. Check if it works as before on the frontend.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/checkbox-list/utils.ts
Comment (thealexandrelara) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/checkbox-list/utils.ts:1]: I'm curious: why is this file necessary? It seems to be empty.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/checkbox-list/style.scss [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+.wc-block-product-filter-checkbox-list__item.hidden {
+	display: none;
+}
+
+
+:where(.wc-block-product-filter-checkbox-list__label) {
+	align-items: center;
+	display: flex;
+	gap: 0.625em;
```
Comment (thealexandrelara) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/checkbox-list/style.scss:15]: Should we use the `$gap` variables in `plugins/woocommerce-blocks/assets/css/abstracts/_variables.scss`?

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/chips/utils.ts
Comment (thealexandrelara) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/chips/utils.ts:1]: This file seems to be empty.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/chips/style.scss
Comment (thealexandrelara) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/chips/style.scss:1]: This file seems to be empty.

**Comment by dinhtungdu** [comment]
@roykho @thealexandrelara This is ready for an early review. I haven't updated E2E tests and fixed lining issues yet. I will do that tomorrow.
As we discussed in the call, we need to preserve the customization made to the display block during the editing session so users can switch between display styles block without losing their settings. I implemented that feature in https://github.com/woocommerce/woocommerce/pull/51096/commits/fedb620818255dbd111bcd68a8507bf0eab3f791 So this PR is ready for an early review.

**Comment by roykho** [comment]
@dinhtungdu for some reason I am getting console errors:

```
blocks.min.js?ver=0d232d232463200f5cfd:19 Block validation: Block validation failed for `woocommerce/product-filter-attribute` ({name: 'woocommerce/product-filter-attribute', icon: {…}, keywords: Array(1), attributes: {…}, providesContext: {…}, …}).

Content generated by `save` function:

<div class="wp-block-woocommerce-product-filter-attribute"></div>

Content retrieved from post body:
```

This is after pulling in the latest and also running `pnpm install` and then `watch:build`

**Comment by dinhtungdu** [comment]
@roykho that's expected as we're updating the block. You're likely to see the message to attempt to recovery, is that correct? Have you reset the Product Filters template part?

**Comment by roykho** [comment]
@dinhtungdu - I didn't see that message prior to the latest commits. Strange. I am not given option to reset the template either.

**Comment by dinhtungdu** [comment]
> I didn't see that message prior to the latest commits. Strange. I am not given option to reset the template either.

@roykho you need to reset the template because the structure is different from the previous one. If the old block structure is used, it's expected to have the error you're getting, which is fine IMO because the block is experimental.

**Comment by roykho** [comment]
> you need to reset the template because the structure is different from the previous one. If the old block structure is used, it's expected to have the error you're getting, which is fine IMO because the block is experimental.

As I mentioned above, I don't see an option to reset.

**Comment by dinhtungdu** [comment]
@roykho the issue you got is related to outdated content of the attribute block, which I fixed in [a373f56](https://github.com/woocommerce/woocommerce/pull/51096/commits/a373f56d90104865dd4841ee5358bd2cec4f509c). You shouldn't see that error anymore.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/components/initial-disabled/index.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * component messes up the focus on inner blocks. For example, when a there is
+ * an heading block nested inside, the text cursor, which indicates the editable
+ * area, isn't visible when focus on the heading block.
+ *
+ * This component only use CSS to control the selecte behavior of inner blocks,
+ * which fix the issues like mentioned above.
+ *
+ * Limitation: being a static component comes with a limitation, this component
+ * is meant to put directly inside the block wrapper element that holds block
+ * props.
```
Comment (thealexandrelara) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/components/initial-disabled/index.tsx]: Just suggesting a few fixes to some phrases in this comment:

```suggestion
/**
 * The reason for this component instead of using core/disabled is, the Disabled
 * component messes up the focus on inner blocks. For example, when there is
 * a heading block nested inside, the text cursor, which indicates the editable
 * area, isn't visible when focus on the heading block.
 *
 * This component only uses CSS to control the selection behavior of inner blocks,
 * which fixes the issues like mentioned above.
 *
 * Limitation: being a static component comes with a limitation, this component
 * is meant to be put directly inside the block wrapper element that holds block
 * props.
```

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/active-filters/block.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
 	"$schema": "https://schemas.wp.org/trunk/block.json",
 	"name": "woocommerce/product-filter-active",
 	"version": "1.0.0",
-	"title": "Filter Options",
+	"title": "Active",
```
Comment (thealexandrelara) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/active-filters/block.json]: Should we use the `Active (Experimental)` as before?

```suggestion
	"title": "Active (Experimental)",
```

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/attribute-filter/block.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
 	"$schema": "https://schemas.wp.org/trunk/block.json",
 	"name": "woocommerce/product-filter-attribute",
 	"version": "1.0.0",
-	"title": "Filter Options",
+	"title": "Attribute",
```
Comment (thealexandrelara) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/attribute-filter/block.json]: Should we use the `Attribute (Experimental)` as before?

```suggestion
	"title": "Attribute (Experimental)",
```

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/components/initial-disabled/index.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	isDisabled: boolean;
+	children: React.ReactNode;
+} ): JSX.Element => (
+	<div className="wc-component-initial-disabled">
+		<div className="wc-component-initial-disabled-overlay" />
+		{ children }
+	</div>
+);
+
+export default InitialDisabled;
```
Comment (thealexandrelara) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/components/initial-disabled/index.tsx]: It seems to be common in our codebase to use named exports when exporting elements from a file, likely because it helps IntelliSense automatically import those elements into other files, and it also makes them easier to search for within the codebase.

```suggestion
export const InitialDisabled = ( {
	children,
}: {
	isDisabled: boolean;
	children: React.ReactNode;
} ): JSX.Element => (
	<div className="wc-component-initial-disabled">
		<div className="wc-component-initial-disabled-overlay" />
		{ children }
	</div>
);
```

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterChips.php [hunk 6]
```diff
@@ -0,0 +1,17 @@
+<?php
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Blocks\BlockTypes;
+
+/**
+ * Product Filter: Attribute Block.
```
Comment (thealexandrelara) [inline plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterChips.php]: I believe this comment should reflect the correct block name.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/checkbox-list/edit.tsx [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+												xmlns="http://www.w3.org/2000/svg"
+											>
+												<path
+													d="M9.25 1.19922L3.75 6.69922L1 3.94922"
+													stroke="currentColor"
+													strokeWidth="1.5"
+													strokeLinecap="round"
+													strokeLinejoin="round"
+												/>
+											</svg>
```
Comment (thealexandrelara) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/checkbox-list/edit.tsx]: I'm wondering if this icon shouldn't be moved to it's own file.

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterCheckboxList.php [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+	/**
+	 * Get the color value from the color attributes.
+	 *
+	 * @param string $key        The key of the color attribute.
+	 * @param array  $attributes The block attributes.
+	 * @return string
+	 */
+	private function get_color_attribute_value( $key, $attributes ) {
+		if ( $attributes[ $key ] ) {
+			return $attributes[ $key ];
```
Comment (thealexandrelara) [inline plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterCheckboxList.php]: Should it be using `esc_attr` here?

```suggestion
			return esc_attr( $attributes[ $key ] );
```

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterCheckboxList.php [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+		}
+
+		if ( $attributes[ 'custom' . ucfirst( $key ) ] ) {
+			return $attributes[ 'custom' . ucfirst( $key ) ];
```
Comment (thealexandrelara) [inline plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterCheckboxList.php]: Same here?

```suggestion
			return esc_attr( $attributes[ 'custom' . ucfirst( $key ) ] );
```

**Review by thealexandrelara — CHANGES_REQUESTED** [review]
Thank you for working on this @dinhtungdu. I left some comments about the code. I noticed that the Price Filter and Stock Status Filter blocks were removed from the template part. Will they be re-added in the future?

I had the same error reported by Roy. I tested in a new JN website.

## Test summary:

❌ Scenario 1: A merchant visits the Product Catalog page in a store without any products or attributes. The merchant adds the Product Filters (Experimental) block to the template.
- Expected Outcome: The Product Filters (Experimental) block is correctly displayed.
- Details: The Attribute filter block is broken when I add it to the Product Catalog template in a store without products or attributes.

https://github.com/user-attachments/assets/81a24fc6-1882-4b42-9cd4-08ea3544c3eb

---


❌ Scenario 2: A merchant visits the Product Catalog page in a store WITH products and attributes. The merchant adds the Product Filters (Experimental) block to the template.
- Expected Outcome: The Product Filters (Experimental) block is correctly displayed.
- Details: The Attribute filter block is broken when I add it to the Product Catalog template in a store with products and attributes.

https://github.com/user-attachments/assets/54066e3f-c006-4a8e-b1f6-dacb11da500a

---

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/components/initial-disabled/index.tsx (hunk 2 above)
Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/components/initial-disabled/index.tsx]: Thanks, I updated the comment per your suggestions.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/components/initial-disabled/index.tsx (hunk 5 above)
Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/components/initial-disabled/index.tsx]: I removed the default export per your suggestion.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/active-filters/block.json (hunk 3 above)
Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/active-filters/block.json]: Yeah, let's do it.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/checkbox-list/edit.tsx (hunk 7 above)
Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/checkbox-list/edit.tsx]: Makes sense, let's do it.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/checkbox-list/style.scss (hunk 1 above)
Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/checkbox-list/style.scss:15]: No, because that variable is based on `$grid-unit` from Core, which is a pixel value. We want to use `em` here so the spacing can adapt to the font size.

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterChips.php (hunk 6 above)
Comment (dinhtungdu) [reply plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterChips.php]: Good catch!

**Review by dinhtungdu — COMMENTED** [review]
@thealexandrelara Thanks so much for the detailed review. I address all your feedback in [`c5e3e3a` (#51096)](https://github.com/woocommerce/woocommerce/pull/51096/commits/c5e3e3a9d312b6a7415185ac13ad51aeb6a5b7e4), can you please review this again?

> I noticed that the Price Filter and Stock Status Filter blocks were removed from the template part. Will they be re-added in the future?

Yes, we need to update those blocks to not use the wrapper. To not extend the scope of this PR too much, I decided to remove it and we will add them back when we work on each.

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterCheckboxList.php (hunk 8 above)
Comment (dinhtungdu) [reply plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterCheckboxList.php]: Good catch. I fixed this issue in 05b9d281b65fe854e9469ff1231fe4130a86122e. But instead of escaping here and in your another comment, I ended up escaping lately in the render method.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/chips/utils.ts
Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/chips/utils.ts:1]: This was fixed in [`ba154cd` (#51096)](https://github.com/woocommerce/woocommerce/pull/51096/commits/ba154cd789bd3c1e1906db8c30bf520a1ec9c6ec)

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/chips/style.scss
Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/chips/style.scss:1]: This was fixed in [`ba154cd` (#51096)](https://github.com/woocommerce/woocommerce/pull/51096/commits/ba154cd789bd3c1e1906db8c30bf520a1ec9c6ec)

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/attribute-filter/edit.tsx [hunk 10]
```diff
@@ -69,144 +76,181 @@ const Edit = ( props: EditProps ) => {
...
+		sortOrder,
+		hideEmpty,
+		isTermsLoading,
+		isFilterCountsLoading,
+	] );
+
+	const { children, ...innerBlocksProps } = useInnerBlocksProps(
+		useBlockProps(),
+		{
+			allowedBlocks: getAllowedBlocks( DISALLOWED_BLOCKS ),
```
Comment (roykho) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/attribute-filter/edit.tsx]: I think this is a bit confusing you're trying to get allowed blocks with a constant that says `DISALLOWED_BLOCKS`

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/utils.ts [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+ */
+import { BlockInstance, getBlockTypes } from '@wordpress/blocks';
+
+/**
+ * Returns an array of allowed block names excluding the disallowedBlocks array.
+ *
+ * @param disallowedBlocks Array of block names to disallow.
+ * @return Array of allowed block names.
+ */
+export const getAllowedBlocks = ( disallowedBlocks: string[] ) => {
```
Comment (roykho) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/utils.ts:12]: I mentioned this above but maybe instead of `disallowedBlocks` we change it to `excludedBlocks`? So it reads like "Get me all the allowed blocks, excluding these ones" instead of "Get me all the allowed blocks disallowing these blocks".

**Review by roykho — CHANGES_REQUESTED** [review]
Thanks for design change and effort! Mostly looking great. I left some small comments.

🛑 I also ran into an issue where the "Clear" button for the Attributes filter not showing up in the frontend when filters are applied. I am referring to this in the editor: 
Google Chrome - Product Catalog ‹ Template ‹ woo-blocks-dev ‹ Editor — WordPress 2024-09-09 at 6 22 46 AM

**Review by thealexandrelara — CHANGES_REQUESTED** [review]
Thank you for working on the suggested changes from my previous review. I left a question regarding the current behaviour of the Attribute filter block registration and I also noticed the same issue reported by Roy.

## Test summary:

✅  Scenario 1: A merchant visits the Product Catalog page in a store without any products or attributes. The merchant adds the Product Filters (Experimental) block to the template.
- Expected Outcome: The Product Filters (Experimental) block is correctly displayed.
- Details: After adding the Product Filters (Experimental) block, it correctly loads and displays the Active filters and the Attribute filter. The block is correctly being displayed on the frontend as well.

---


✅  Scenario 2: A merchant visits the Product Catalog page in a store WITH products and attributes. The merchant adds the Product Filters (Experimental) block to the template.
- Expected Outcome: The Product Filters (Experimental) block is correctly displayed.
- Details: After adding the Product Filters (Experimental) block, it correctly loads and displays the Active filters and the Attribute filter. The block is correctly being displayed on the frontend as well.

--- 

✅ Scenario 3: Merchants can easily edit the individual filter blocks settings and options.
- Conditions:
    - A merchant visits the Product Catalog template. 
    - The Product Catalog template contains a Product Filters (Experimental) block.
    - The Merchant clicks on the options for an individual filter.
- Expected Outcome: The individual filter is selected (not the list block within it).
- Details: Once the merchant clicks on the list of options for an individual filter, that filter is selected and the Inspector Control correctly displays the block settings for the merchant to edit. If the settings are modified, the canvas displays the changes. By saving the template and visiting the frontend page, we can ensure the changes are being applied.

---

❓  Scenario 4: A merchant visits the Product Catalog template and decides to add another Attribute filter within the Product Filters (Experimental) block.
- Expected Outcome: The merchant can use the block inserter to add a new Attribute filter block.
- Details: I noticed that the Attribute filter block itself is not available; however, it's variations are (the Size and Color variations). Is that expected?

CleanShot 2024-09-09 at 15 44 16@2x

---

❌  Scenario 5: Shoppers can see the "Clear" button next to the filter heading.
- Conditions:
    - A shoppers visits the Shop page (/page) that contains a Product Filters (Experimental) block.
    - The shopper selects a filter option from the list.
- Expected Outcome: The "Clear" button should appear next to the Heading block.
- Details: Roy mentioned this above, and I'm also facing the same issue, when a filter option is selected I'm not seing the "Clear" button that allows shoppers to remove the applied filters.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/utils.ts (hunk 11 above)
Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/utils.ts:12]: Make sense! I renamed the constant followed your suggestion. Thanks!

**Comment by dinhtungdu** [comment]
> ❓ Scenario 4: A merchant visits the Product Catalog template and decides to add another Attribute filter within the Product Filters (Experimental) block.
> Expected Outcome: The merchant can use the block inserter to add a new Attribute filter block.
Details: I noticed that the Attribute filter block itself is not available; however, it's variations are (the Size and Color variations). Is that expected?

@thealexandrelara It's expected with the current implementation, as I set the first variation of the attribute filter as default, which will hide the attribute filter block from the inserter. I did it because we have a default attribute, so even if we keep the attribute block in the inserter, after the block is added to the editor, it still renders the default attributes options. I don't have a strong opinion on this, happy to remove that if you think we should keep the Attribute block in the inserter.

**Comment by dinhtungdu** [comment]
@roykho @thealexandrelara thanks for the review. I addressed your feedback and updated the e2e test too. Please give this one another look.

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterCheckboxList.php [hunk 12]
```diff
@@ -71,7 +71,7 @@ function ( $item ) {
 		ob_start();
 		?>
 		<div <?php echo get_block_wrapper_attributes( $wrapper_attributes ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>>
-			<ul class="wc-block-product-filter-checkbox-list__list">
+			<ul class="wc-block-product-filter-checkbox-list__list" aria-label="<?php echo esc_attr( 'Filter Options', 'woocommerce' ); ?>">
```
Comment (roykho) [inline plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterCheckboxList.php]: This should be `esc_attr__()` or remove the echo and use `esc_attr_e()`

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/utils.ts (hunk 11 above)
Comment (roykho) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/utils.ts:12]: I think we should change the argument here as well to be `excludedBlocks`. Also I think this utility function already exists here: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/product-filter/utils.ts

So we should consolidate it.

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterCheckboxList.php (hunk 12 above)
Comment (dinhtungdu) [reply plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterCheckboxList.php]: I already fixed it in [9b632cb](https://github.com/woocommerce/woocommerce/pull/51096/commits/9b632cb5184e30b990f29691819c035ef05444f8).

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/utils.ts (hunk 11 above)
Comment (roykho) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/utils.ts:12]: @dinhtungdu I am not sure you saw this comment?

**Review by thealexandrelara — CHANGES_REQUESTED** [review]
> @thealexandrelara It's expected with the current implementation, as I set the first variation of the attribute filter as default, which will hide the attribute filter block from the inserter. I did it because we have a default attribute, so even if we keep the attribute block in the inserter after the block is added to the editor, it still renders the default attribute options. I don't have a strong opinion on this, happy to remove that if you think we should keep the Attribute block in the inserter.

I see. I'm just thinking about how it would be if there were 30 attributes or more in a store, and the merchant did not know precisely which attribute they wanted to add. I would probably type "Attribute" in the block inserter to add the attribute filter block and select one from the list of options.

I noticed some issues with the current version, for some reason the "Close" button is getting rendered when I add the template part to the template.


## Test summary:

⚠️  Scenario 1: A merchant visits the Product Catalog page in a store without any products or attributes. The merchant adds the Product Filters (Experimental) block to the template.
- Expected Outcome: The Product Filters (Experimental) block is correctly displayed.
- Details: Unfortunately, for some reason, the Product Filters Overlay Navigation block is rendering the "Close" button when I add the Product Filters (Experimental) template part to the template.

---


⚠️  Scenario 2: A merchant visits the Product Catalog page in a store WITH products and attributes. The merchant adds the Product Filters (Experimental) block to the template.
- Expected Outcome: The Product Filters (Experimental) block is correctly displayed.
- Details: Unfortunately, for some reason, the Product Filters Overlay Navigation block is rendering the "Close" button when I add the Product Filters (Experimental) template part to the template.


https://github.com/user-attachments/assets/b32a9a1b-1340-46c5-a01c-c9b37405e007

--- 

✅ Scenario 3: Merchants can easily edit the individual filter blocks settings and options.
- Conditions:
    - A merchant visits the Product Catalog template. 
    - The Product Catalog template contains a Product Filters (Experimental) block.
    - The Merchant clicks on the options for an individual filter.
- Expected Outcome: The individual filter is selected (not the list block within it).
- Details: Once the merchant clicks on the list of options for an individual filter, that filter is selected, and the Inspector Control correctly displays the block settings for the merchant to edit. If the settings are modified, the canvas displays the changes. By saving the template and visiting the frontend page, we can ensure the changes are being applied.

---

✅   Scenario 4: A merchant visits the Product Catalog template and decides to add another Attribute filter within the Product Filters (Experimental) block.
- Expected Outcome: The merchant can use the block inserter to add a new Attribute filter block.
- Details: According to your comment above, only the block variations are expected to appear in the block inserter. I believe we should also have the "Attribute" filter available as a way to allow merchants to always have a block they can rely on.

---

✅  Scenario 5: Shoppers can see the "Clear" button next to the filter heading.
- Conditions:
    - A shoppers visits the Shop page (/page) that contains a Product Filters (Experimental) block.
    - The shopper selects a filter option from the list.
- Expected Outcome: The "Clear" button should appear next to the Heading block.
- Details: When shoppers select an option from the attributes filter block, the clear button is displayed, and they can click on it to clear the selected options for that individual filter.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/utils.ts (hunk 11 above)
Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/utils.ts:12]: @roykho Sorry I missed this. We will eventually remove the `product-filter` block, so I didn't pay too much attention to it. I kept the `product-filter` block for the migration purpose, but it has done its job now, let me remove it to prevent confusion.

**Comment by dinhtungdu** [comment]
@thealexandrelara the first and second test cases are not under the scope of this PR. This PR doesn't touch the navigation button, the template part is brought over from the trunk, and I only modified the filter block part.

With #50273 is now done, I updated the template part to remove the overlay navigation button in [b39cf76](https://github.com/woocommerce/woocommerce/pull/51096/commits/b39cf7613539f6d8fd847a1540ad760ab31a65b5) as the default overlay setting is Never.

**Review by thealexandrelara — APPROVED** [review]
Great work! Thank you for working on this, Tung! 🚀

**Review by roykho — APPROVED** [review]
Thanks for working on this and all the updates! 🚢 

✅ The Attribute Filter block picks up the correct block to render, with the correct title set to the product attribute label.
✅ Selecting one attribute, see the selected attribute appears in the Active block
✅ See the Product Collection block updates.
✅ See the URL updates.
✅ See the preview renders Active and Attribute filter block.
✅ From the canvas, click on the filter options. See the Attribute Filter block is selected, not the List block.
✅ Try changing the block setting, see the preview updates.
✅ Click on the filter options again, and see the List block is selected.
✅ Change the color settings of the List block, see the editor preview update as you change.
✅ Change the display style from List to Chip. See the editor preview change
✅ Change back to the List block, see the customization you did previously is preserved.
✅ Save the template. Check if it works as before on the frontend
